test/utils_tests.js

```diff
@@ -108,5 +108,72 @@ describe('Utils', function () {
             assert.equal(best[0].element.id, 'fat');
         });
 
+        it('test non-attribute', function () {
+            const doc = staticDom(`
+                <img id = "foo" alt = "bat"></img><img id ="bar"></img>
+            `);
+            const rules = ruleset(
+                rule(dom('img'), type('attr')),
+                rule(type('attr'), score(scoreFunc)),
+                rule(type('attr').max(), out('best'))
+            );
+
+            function scoreFunc(fnode) {
+                return searchAttributes(fnode, searchFunc, 'alt')? 5 : 1;
+            }
+
+            function searchFunc(attr) {
+                return attr.includes('at');
+            }
+            const facts = rules.against(doc);
+            const best = facts.get('best');
+            assert.equal(best.length, 1);
+            assert.equal(best[0].element.id, 'foo');
+        });
+
+        it('test negative', function () {
+            const doc = staticDom(`
+                <img id = "foo"></img><img id ="bar"></img>
+            `);
+            const rules = ruleset(
+                rule(dom('img'), type('attr')),
+                rule(type('attr'), score(scoreFunc)),
+                rule(type('attr').max(), out('best'))
+            );
+
+            function scoreFunc(fnode) {
+                return searchAttributes(fnode, searchFunc)? 5 : 1;
+            }
+
+            function searchFunc(attr) {
+                return attr.includes('z');
+            }
+            const facts = rules.against(doc);
+            const best = facts.get('best');
+            assert.equal(best.length, 2);
```

Comment:
Is this `2` because nothing matches and therefore are all "the best" in that they are all equal?

Reply:
yes!